src/device.jl

```diff
@@ -18,9 +25,11 @@ function device_reset(dev::Integer)
     for p in todelete
         delete!(cuda_ptrs, p)
     end
-    # Reset the device
-    device(dev)
-    rt.cudaDeviceReset()
+
+    # no need to reset the entire device, just reset the primary context
+    pctx = CuPrimaryContext(dev)
+    CUDAdrv.reset(pctx)
```

Comment:
Hmm. According to the docs, it should be safe to do a `Release` after a `Reset`, so I didn't bother. I had also hoped other calls would generate a error `context not initialized` instead of segfaulting...